core/primitives/src/errors.rs

```diff
@@ -12,7 +12,7 @@ use near_vm_errors::VMError;
 #[derive(
     BorshSerialize, BorshDeserialize, Debug, Clone, PartialEq, Eq, Deserialize, Serialize, RpcError,
 )]
-#[rpc_error_parent = "ServerError"]
+#[rpc_error_variant = "TxExecutionError"]
```

Comment:
What is the exact point of this marker? If this is just to rename the struct, I vote for the enum renaming as it would certainly benefit from disambiguation with the VM execution error.

Reply:
We rely on the serde json serialization, it uses enum Variant name as an identifier (of the error type in our case). By parsing Enum, we collect all the "subtypes" of the error (in this case it will correlate to `ServerError::TxExecutionError`). We could just force to use the same identifier for our "subtypes" as variant name, but I guess it'll be less clear.